lib/Cake/Controller/Component/Auth/BasicAuthenticate.php

```diff
@@ -82,8 +82,17 @@ public function authenticate(CakeRequest $request, CakeResponse $response) {
  * @return mixed Either false or an array of user information
  */
 	public function getUser(CakeRequest $request) {
-		$username = env('PHP_AUTH_USER');
-		$pass = env('PHP_AUTH_PW');
+		$username = '';
+		$pass = '';
+		if (isset($_SERVER['PHP_AUTH_USER'])) {
```

Comment:
This `if..else` is not necessary. Keep the existing statements on line 85-86 and just try to fetch the credentials from headers inside `if (empty($username)) {` block.